lib/members_page.rb

```diff
@@ -0,0 +1,91 @@
+# encoding: utf-8
+# frozen_string_literal: true
+require 'scraped'
+require_relative './decorators/party_name_attributes'
+
+class MembersPage < Scraped::HTML
+  decorator Scraped::Response::Decorator::AbsoluteUrls
+  decorator PartyNameAttributes
+
+  field :party_groupings do
+    noko.xpath('//td/table/tr/td[contains(.,"Circunscripción")]/parent::tr')
+        .map do |row|
+      fragment row => PartyGroupings
+    end
+  end
+
+  class PartyGroupings < Scraped::HTML
+    field :party do
+      party_image_node[:party_name]
+    end
+
+    field :party_id do
+      party_image_node[:party_id]
+    end
+
+    field :members do
+      # 1. Reject image rows
+      # 2. Slice list to separate from other party groupings
+      # 3. First item is member rows in required party group
+      noko.xpath('following-sibling::tr')
+          .reject { |row| row.at_xpath('./td/@colspan').to_s >= '3' }
+          .slice_before { |i| i.text == noko.text }
+          .first.map do |row|
+        fragment row => MemberRow
+      end
+    end
+
+    private
+
+    def party_image_node
+      noko.xpath('preceding-sibling::tr/td/img').last
+    end
+  end
+
+  class MemberRow < Scraped::HTML
+    field :id do
+      source.to_s[/dipt=(\d+)/, 1]
+    end
+
+    field :sort_name do
+      noko.xpath('td').first.text.tidy.sub(/^\d+\s*/, '').sub(' (LICENCIA)', '')
+    end
+
+    field :source do
+      noko.xpath('td/a/@href').text
+    end
+
+    field :area do
+      "#{area_name} #{district_number}"
+    end
+
+    field :area_id do
+      "ocd-division/country:mx/entidad:#{area_part}/#{formatted_district_number}"
+    end
+
+    private
+
+    def area_name
+      noko.xpath('td')[1].text
+    end
+
+    def area_part
+      area.split(' ').reject { |x| x =~ /\d/ }.join('-').downcase
+    end
+
+    def district
+      noko.xpath('td')[2].text
+    end
+
+    def district_number
+      district.split(' ').last
+    end
+
+    def formatted_district_number
```

Comment:
as well as being misleading by implying it's only about districts, saying "formatted" doesn't really convey much here. Anyone reading this is instantly going to say "Formatted *how*?"